pwa/src/components/utility/selectInput.tsx

```diff
@@ -2,45 +2,52 @@ import * as _ from "lodash";
 import * as React from "react";
 
 interface SelectInputProps {
-  options: Array<Record<"value" | "name", any>>;
+  options: Array<Partial<Record<"value" | "name" | "id" | "selected", any>>>;
   name: string;
   nameOverride?: string;
-  id: string;
+  value?: string;
   data?: string;
+  id: string;
   required?: boolean;
+  onChange?: any;
+  disabled?: boolean;
 }
 
+
 /**
  * This component generates a select input.
  * @returns Jsx of the generated form.
  */
-export const SelectInputComponent = (props: SelectInputProps) => {
+export const SelectInputForMultiSelect: React.FC<SelectInputProps> = ({
```

Comment:
When the file is called `SelectInput.tsx`, the component should be called `SelectInput`.